### PR TITLE
fixed excessive outputs copying (in case when the fallback happened) and updated the test for that

### PR DESCRIPTION
**Pull request by myshevts**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/auto_batch/auto_batch.hpp [hunk 1]
```diff
@@ -106,6 +106,7 @@ class AutoBatchInferRequest : public InferenceEngine::IInferRequestInternal {
     void CopyOutputsIfNeeded();
     AutoBatchExecutableNetwork::WorkerInferRequest& _myBatchedRequestWrapper;
     std::exception_ptr _exceptionPtr;
+    bool _wasBatchedRequestUsed;
```
Comment (apankratovantonp) [inline src/plugins/auto_batch/auto_batch.hpp:109]: `= false;`

Comment (myshevts) [reply src/plugins/auto_batch/auto_batch.hpp:109]: Why not true?
Both values  are incorrect as initial

Comment (apankratovantonp) [reply src/plugins/auto_batch/auto_batch.hpp:109]: Just add default initialization.

Comment (myshevts) [reply src/plugins/auto_batch/auto_batch.hpp]: moved to enum to explicitly have un-initialized state

**Review by apankratovantonp — APPROVED** [review]

**Pull request merged**